src/oscar/apps/dashboard/catalogue/forms.py

```diff
@@ -300,25 +300,26 @@ class ProductImageForm(forms.ModelForm):
 
     class Meta:
         model = ProductImage
-        fields = ['product', 'original', 'caption']
+        fields = ['product', 'original', 'caption', 'display_order']
         # use ImageInput widget to create HTML displaying the
         # actual uploaded image and providing the upload dialog
         # when clicking on the actual image.
         widgets = {
             'original': ImageInput(),
+            'display_order': forms.HiddenInput(),
         }
 
-    def save(self, *args, **kwargs):
-        # We infer the display order of the image based on the order of the
-        # image fields within the formset.
-        kwargs['commit'] = False
-        obj = super().save(*args, **kwargs)
-        obj.display_order = self.get_display_order()
-        obj.save()
-        return obj
+    def __init__(self, data=None, *args, **kwargs):
+        self.prefix = kwargs.get('prefix', None)
+        instance = kwargs.get('instance', None)
+        if not instance:
+            initial = {'display_order': self.get_display_order()}
+            initial.update(kwargs.get('initial', {}))
+            kwargs['initial'] = initial
+        super(ProductImageForm, self).__init__(data, *args, **kwargs)
```

Comment:
Please use `super()` without any arguments, for consistency across codebase now that we are on Python 3. This means we only use the long form where we are explicitly calling super on something other than the parent class.

Reply:
This and above super calls left from original implementation. Fixing.